### PR TITLE
Heretic Themed Toys

### PR DESCRIPTION
**Pull request by eeSPee**
## About The Pull Request

Adds fake tears and codex toys, and adds them to the arcade toy list.

Ported from https://github.com/BeeStation/BeeStation-Hornet/pull/2756

## Why It's Good For The Game

They're really annoying.

## Changelog
:cl:
add: Codex Cicatrix and Fake Pierced Reality toys.
/:cl:

### CODE REVIEW

File: code/game/objects/items/toys.dm [hunk 1]
```diff
@@ -1495,3 +1495,43 @@
...
 		cooldown = (world.time + 10)
 		addtimer(CALLBACK(GLOBAL_PROC, .proc/playsound, src, 'sound/effects/blobattack.ogg', 50, FALSE), 0.5 SECONDS)
+
+
+/*
+ * Eldrich Toys
+ */
+
+/obj/item/toy/eldrich_book
+	name = "Codex Cicatrix"
```
Comment (Fikou) [inline code/game/objects/items/toys.dm:1505]: no funny name, cringe

File: code/game/objects/items/toys.dm [hunk 2]
```diff
@@ -1495,3 +1495,43 @@
...
+	icon = 'icons/effects/eldritch.dmi'
+	icon_state = "pierced_illusion"
+
+/obj/item/storage/box/heretic_box
+	name = "box of pierced realities"
+	desc = "A box containing toys resembling pierced realities."
+
+/obj/item/storage/box/heretic_box/PopulateContents()
+	for(var/i in 1 to rand(1,4))
+		new /obj/item/toy/reality_pierce(src)
```
Comment (Rohesie) [inline code/game/objects/items/toys.dm:1537]:
```suggestion
		new /obj/item/toy/reality_pierce(src)

```
The linter doesn't like files ending without a line break.

File: code/game/objects/items/toys.dm [hunk 3]
```diff
@@ -1495,3 +1495,43 @@
...
+
+/obj/item/toy/eldrich_book
+	name = "Codex Cicatrix"
+	desc = "A toy book that closely resembles the Codex Cicatrix. Covered in fake polyester human flesh and has a huge goggly eye attached to the cover. The runes are gibberish and cannot be used to summon demons... Hopefully?"
+	icon = 'icons/obj/eldritch.dmi'
+	icon_state = "book"
+	w_class = WEIGHT_CLASS_SMALL
+	attack_verb_continuous = list("sacrifices", "transmutes", "graspes", "curses")
+	attack_verb_simple = list("sacrifice", "transmute", "grasp", "curse")
+	var/open = FALSE
```
Comment (Rohesie) [inline code/game/objects/items/toys.dm:1512]: Give your new variable a short documentation on what it does, what it is about or for.
This one should be rather simple.

**Review by Rohesie — CHANGES_REQUESTED** [review]

File: code/game/objects/items/toys.dm [hunk 4]
```diff
@@ -1495,3 +1495,43 @@
...
+
+/obj/item/toy/eldrich_book/attack_self(mob/user)
+	open = !open
+	update_icon()
+
+/obj/item/toy/eldrich_book/update_icon()
```
Comment (Jared-Fogle) [inline code/game/objects/items/toys.dm:1518]:
```suggestion
/obj/item/toy/eldrich_book/update_icon_state()
```

**Comment by Cartographer-D** [comment]
While you're at it, can you add a heretic action figure to round out the collection?

**Comment by eeSPee** [comment]
Thanks for the feedback!

> While you're at it, can you add a heretic action figure to round out the collection?

If someone does the sprite, yes!

File: code/game/objects/items/toys.dm [hunk 5]
```diff
@@ -1495,3 +1495,43 @@
...
+
+/obj/item/toy/eldrich_book
+	name = "Codex Cicatrix"
+	desc = "A toy book that closely resembles the Codex Cicatrix. Covered in fake polyester human flesh and has a huge goggly eye attached to the cover. The runes are gibberish and cannot be used to summon demons... Hopefully?"
+	icon = 'icons/obj/eldritch.dmi'
+	icon_state = "book"
+	w_class = WEIGHT_CLASS_SMALL
+	attack_verb_continuous = list("sacrifices", "transmutes", "graspes", "curses")
+	attack_verb_simple = list("sacrifice", "transmute", "grasp", "curse")
+	var/book_state = FALSE	//helps determine the icon state of this item when it's used on self
```
Comment (Rohesie) [inline code/game/objects/items/toys.dm:1512]:
```suggestion
	attack_verb_simple = list("sacrifice", "transmute", "grasp", "curse")
	/// Helps determine the icon state of this item when it's used on self.
	var/book_state = FALSE
```
<https://github.com/tgstation/tgstation/blob/master/.github/AUTODOC_GUIDE.md>

File: code/game/objects/items/toys.dm [hunk 6]
```diff
@@ -1495,3 +1495,44 @@
 	if(cooldown <= world.time)
 		cooldown = (world.time + 10)
 		addtimer(CALLBACK(GLOBAL_PROC, .proc/playsound, src, 'sound/effects/blobattack.ogg', 50, FALSE), 0.5 SECONDS)
+
+
+/*
+ * Eldrich Toys
+ */
+
+/obj/item/toy/eldrich_book
```
Comment (MrMelbert) [inline code/game/objects/items/toys.dm:1504]: i don't know if there's a joke i'm missing but i'm pretty sure it's **eldritch** not eldrich

**Comment by Rohesie** [comment]
Test it locally and unset it from draft when you are done, guaranteeing it works as intended.

File: code/game/objects/items/toys.dm (hunk 6 above)
Comment (eeSPee) [reply code/game/objects/items/toys.dm:1504]: typo

**Comment by eeSPee** [comment]
Right, fixed the typo and everything working as intended.

File: code/game/objects/items/toys.dm [hunk 7]
```diff
@@ -1495,3 +1495,44 @@
...
+/obj/item/toy/eldritch_book
+	name = "Codex Cicatrix"
+	desc = "A toy book that closely resembles the Codex Cicatrix. Covered in fake polyester human flesh and has a huge goggly eye attached to the cover. The runes are gibberish and cannot be used to summon demons... Hopefully?"
+	icon = 'icons/obj/eldritch.dmi'
+	icon_state = "book"
+	w_class = WEIGHT_CLASS_SMALL
+	attack_verb_continuous = list("sacrifices", "transmutes", "graspes", "curses")
+	attack_verb_simple = list("sacrifice", "transmute", "grasp", "curse")
+	/// Helps determine the icon state of this item when it's used on self.
+	var/book_state = FALSE
```
Comment (Jared-Fogle) [inline code/game/objects/items/toys.dm:1513]: This variable name is not descriptive, rename it to something that explains that it is whether the book is *open*, not just a "state", which doesn't even suggest a boolean.

**Review by Mothblocks — CHANGES_REQUESTED** [review]

File: code/game/objects/items/toys.dm (hunk 7 above)
Comment (Jared-Fogle) [reply code/game/objects/items/toys.dm:1513]: Funnily enough, it looks like you had it right before with `open`! 😄

**Comment by tgstation-server** [comment]
Your Fix/Feature pull request delta is currently below zero (-2). Maintainers may close future Feature/Tweak/Balance PRs. Fixing issues or helping to improve the codebase will raise this score.

**Pull request merged**